src/humanloop/otel/exporter.py

```diff
@@ -0,0 +1,224 @@
+import logging
+import typing
+from queue import Queue
+from threading import Thread
+from typing import Any, Optional
+
+from opentelemetry import trace
+from opentelemetry.sdk.trace import ReadableSpan
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+
+from humanloop.otel.constants import HL_FILE_OT_KEY, HL_LOG_OT_KEY, HL_OT_EMPTY_VALUE, HL_TRACE_METADATA_KEY
+from humanloop.otel.helpers import is_humanloop_span, read_from_opentelemetry_span
+from humanloop.requests.flow_kernel_request import FlowKernelRequestParams
+from humanloop.requests.prompt_kernel_request import PromptKernelRequestParams
+
+if typing.TYPE_CHECKING:
+    from humanloop.base_client import BaseHumanloop
+
+
+logger = logging.getLogger("humanloop.sdk")
+
+
+class HumanloopSpanExporter(SpanExporter):
+    """Upload Spans created by SDK decorators to Humanloop.
+
+    Spans not created by Humanloop SDK decorators will be ignored.
+    """
+
+    DEFAULT_NUMBER_THREADS = 4
```

Comment:
should this be user configurable? What concurrency are you doing here?

Reply:
Uploading spans in parallel. There is a bottleneck on Flow Traces since with our current backend Flow Traces must be uploaded top-to-bottom, one-at-a-time. There is an open question in the PR description for implementing a new endpoint that allows uploading the Trace at once.

Otherwise, the gain in speed happens when there are multiple Spans whose parent has been uploaded e.g. a Flow Span that has N direct Prompt Spans as children (this corresponds to a conversation agent for example)

Reply:
> There is an open question in the PR description for implementing a new endpoint that allows uploading the Trace at once.

This would be very helpful in general

Reply:
Should we do it to support this PR or scope it out?

Reply:
can be a fast follow i suspect, not blocking, but should be scoped out so we don't forget